### ISSUE TITLE
[Question] How to sample an event time delta range within [0, 10,000]

### ISSUE THREAD

**Issue by SiriusHou** [ant-research/EasyTemporalPointProcess#55]
In your example data, `time_since_last_event` is always within the range [0, 10]. If my sampled `time_since_last_event` can range from [0, 10,000], can you guide me on how to sample it?

**Comment by iLampard** [comment]
Hi， this is a 'dtime_max' in the thinning algo params that determine the range of the 

```
model_config:
.....

    thinning:

      .....

      dtime_max: 5.    <-------------------- HERE
....

```

**Comment by SiriusHou** [comment]
Thank you for your answer. In fact, I adjusted this dtime_max but it didn't help.

**Comment by SiriusHou** [comment]
Even after adjusting `dtime_max` in reproducing retweet results #49, the event type prediction accuracy remains lower than when normalizing the data delta time to the range [0, 10].

**Comment by iLampard** [comment]
let me have a look

**Comment by SiriusHou** [comment]
I'm not sure I understand your code correctly. [Here](https://github.com/ant-research/EasyTemporalPointProcess/blob/9e9bc4e6d864055793aeda7f5ed92fb51e8d2cc5/easy_tpp/preprocess/event_tokenizer.py#L408) I found you used `pad_token_id` to pad `time_delta_sequence`. Suppose we have 10 event types, but the delta time can be as large as 100. Should we use 100 to pad the `time_delta_sequence`?

**Comment by iLampard** [comment]
> I'm not sure I understand your code correctly. [Here](https://github.com/ant-research/EasyTemporalPointProcess/blob/9e9bc4e6d864055793aeda7f5ed92fb51e8d2cc5/easy_tpp/preprocess/event_tokenizer.py#L407) I found you used `pad_token_id` to pad `time_delta_sequence`. Suppose we have 10 event types, but the delta time can be as large as 100. Should we use 100 to pad the `time_delta_sequence`?

Hi, 

The perfect case is indeed to use a different pad token for time_delta_sequence. 

The current implementation of using type pad token is a simple workaround. When computing loss, we use masks from type_sequence to eliminate padded events, and therefore, the pad tokens for time_delta_sequence are not used.  

see https://github.com/ant-research/EasyTemporalPointProcess/blob/main/easy_tpp/model/torch_model/torch_basemodel.py#L110


Another reason is there is not a straightforward way to determine the pad token id for time sequences. One way is to compute the statistics of the time delta sequences and then choose a large number. But this causes computations and not very friendly for users.

**Comment by iLampard** [comment]
Following up on this after digging into the code — your observations were exactly right, and here is what was actually happening:

**Why adjusting `dtime_max` didn't help:** it doesn't control the sampling horizon at all. Inside the thinning sampler, `dtime_max` is only used as the *fallback value* assigned when all proposals get rejected ([torch_thinning.py#L163](https://github.com/ant-research/EasyTemporalPointProcess/blob/main/easy_tpp/model/torch_model/torch_thinning.py#L163)); the `dtime_boundary` computed from it is passed into `draw_next_time_one_step` but never used. The effective horizon is implicit — `num_exp` exponential draws at the estimated intensity upper bound.

**Why normalizing to [0, 10] fixes accuracy:** the models themselves are not scale-invariant. With raw times in the thousands, the sin/cos temporal encodings (THP/AttNHP) alias, `nn.Linear(1, hidden)` time embeddings (RMTPP/ANHN) saturate downstream nonlinearities, and NHP's `exp(-delta * dt)` decay collapses to zero — so the hidden states the mark classifier reads are degenerate. That's why *type* accuracy drops too. Rescaling is the correct practice (most TPP benchmark datasets are pre-scaled), not a workaround.

**This is now built in.** As of 6531a19 on main, you can enable opt-in time rescaling in the data config:

```yaml
data:
  mydata:
    ...
    data_specs:
      num_event_types: 10
      pad_token_id: 10
      rescale_time: true      # divide all times by the mean training inter-event time (mean dt ~= 1)
      # time_scale: 1000.0    # optionally set an explicit divisor instead
```

Time predictions and RMSE are automatically converted back to your original time units, and the resolved scale is persisted in the model config. Default behavior is unchanged when the flag is off (existing results remain reproducible); a startup warning now suggests the option when inter-event times look large. It will be included in the next PyPI release.

(Also worth noting: `IntensityFree` handles scale through its log-space standardization, which had a separate bug fixed in #84 / v0.2.3.)

**Comment by iLampard** [comment]
The `rescale_time` option shipped in [easy-tpp 0.2.4](https://github.com/ant-research/EasyTemporalPointProcess/releases/tag/0.2.4) — `pip install -U easy-tpp`. Closing; feel free to reopen if anything does not work as expected.
